atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTableTest.java

```diff
@@ -225,25 +218,158 @@ public void dropTablesDoesNotThrowIfMultipleMatchingTableReferencesExistAndCaseS
             throws TableMappingNotFoundException {
         // This mock is never called on the case sensitive code path, but this makes the test easier to read.
         lenient().when(sqlConnection.selectCount(any(), any(), any())).thenReturn((long) 2);
-        createTable();
-        createOverflowTable();
+        createTableAndOverflow();
 
         assertThatCode(() -> tableMappingDdlTable.drop()).doesNotThrowAnyException();
     }
 
+    @Test
+    public void createAltersTableIfConfiguredAndOverflowTableExists() throws TableMappingNotFoundException {
+        createTableAndOverflow();
+        setTableToHaveOverflowColumn(false);
+        setMetadataToHaveOverflow(true);
+        tableMappingDdlTable.create(createMetadata(true));
+        verifyNumberOfTimesTableAltered(1);
+    }
+
+    @Test
+    public void createDoesNothingWhenAlterSpecifiedButOverflowColumnExists() throws TableMappingNotFoundException {
+        createTableAndOverflow();
+        setTableToHaveOverflowColumn(true);
+        setMetadataToHaveOverflow(true);
+        assertThatCode(() -> tableMappingDdlTable.create(createMetadata(true))).doesNotThrowAnyException();
+        verifyNumberOfTimesTableAltered(0);
+    }
+
+    @Test
+    public void createDoesNothingWhenAlterSpecifiedButMigrationUnstarted() throws TableMappingNotFoundException {
```

Comment:
nit: add a test for when the overflow table does not exist :) (which is subtly different from a table mapping not existing)